pkgs/desktops/cinnamon/xviewer/default.nix

```diff
@@ -0,0 +1,74 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, autoreconfHook
+, cinnamon-desktop
+, file
+, gdk-pixbuf
+, glib
+, gobject-introspection
+, gtk-doc
+, gtk3
+, intltool
+, itstool
+, lcms2
+, libexif
+, libjpeg
+, libpeas
+, libtool
+, libxml2
+, pkg-config
+, shared-mime-info
+, wrapGAppsHook
+, xapps
+, yelp-tools }:
+
+stdenv.mkDerivation rec {
+  pname = "xviewer";
+  version = "2.8.3";
+
+  src = fetchFromGitHub {
+    owner = "linuxmint";
+    repo = pname;
+    rev = version;
+    sha256 = "0h3qgqaiz5swy09fr6z3ag2952hgzsk5d2fpwmwb78yjrzrhnzpy";
+  };
+
+  nativeBuildInputs = [
+    wrapGAppsHook
+    autoreconfHook
+    cinnamon-desktop
+    gdk-pixbuf
+    gobject-introspection
+    gtk-doc
+    intltool
+    itstool
+    libtool
+    pkg-config
+    yelp-tools
+  ];
+
+  buildInputs = [
+    glib
+    gtk3
+    libexif
+    libjpeg
+    libpeas
+    libxml2
+    shared-mime-info
+    xapps
+    lcms2
+  ];
+
+  postAutoreconf = ''
+    substituteInPlace configure --replace /usr/bin/file "${file}/bin/file"
```

Comment:
I noticed in other builds that their `./configure` complains about `file` not being found, probably because these are missing a patch like yours. But is this patch actually necessary? (Probably, as else you wouldn't have taken the trouble to include it in the build.) Could you add a short one-line comment?

Reply:
It is probably not necessary. I just saw configure tried to use file and precautiously patched it. If that happens in other builds to maybe this is something to be patched in autoconf/automake?